src/cache.cpp

```diff
@@ -786,6 +786,10 @@ bool convert_header_to_stat(const char* path, const headers_t& meta, struct stat
     if(pst->st_mtime < 0){
         pst->st_mtime = 0L;
     }else{
+        if(mtime.tv_sec < 0){
+            mtime.tv_sec  = 0;
+            mtime.tv_nsec = 0;
```

Comment:
Can you add a test to `test_read_external_object`?

Reply:
Yes, I added